parser_library/src/processing/instruction_sets/asm_processor.cpp

```diff
@@ -607,4 +611,39 @@ context::id_index asm_processor::find_sequence_symbol(const rebuilt_statement& s
     }
 }
 
+namespace {
+class AINSERT_operand_visitor final : public expressions::mach_expr_visitor
+{
+public:
+    // Inherited via mach_expr_visitor
+    void visit(const expressions::mach_expr_constant&) override {}
+    void visit(const expressions::mach_expr_data_attr&) override {}
+    void visit(const expressions::mach_expr_symbol& expr) override { value = expr.value; }
+    void visit(const expressions::mach_expr_location_counter&) override {}
+    void visit(const expressions::mach_expr_self_def&) override {}
+    void visit(const expressions::mach_expr_default&) override {}
+
+    context::id_index value = nullptr;
+};
+} // namespace
+
+void asm_processor::process_AINSERT(rebuilt_statement stmt)
+{
+    const auto& ops = stmt.operands_ref();
+
+    if (!check(stmt, hlasm_ctx, checker_, *this))
+        return;
+
+    const auto& record = dynamic_cast<const semantics::string_assembler_operand&>(*ops.value[0]).value;
+    AINSERT_operand_visitor visitor;
+    dynamic_cast<const semantics::expr_assembler_operand&>(*ops.value[1]).expression->apply(visitor);
```

Comment:
Would it be enough to just dynamic_cast the machine expression to expressions::mach_expr_symbol and get the value from there? Seems easier than defining a visitor.